firewall/filesystem_storage_firewall.go

```diff
@@ -0,0 +1,41 @@
+package firewall
+
+import (
+	"errors"
+)
+
+type FilesystemAcraFirewall struct {
+
+	handlers []QueryHandler
+
+	storageDirectory string
+}
+
+
+var ErrHandlersNotSpecified = errors.New("at least one query handler should be specified")
+
+
+func NewFilesystemAcraFirewall(storageDirectory string) (*FilesystemAcraFirewall, error) {
+
+	return &FilesystemAcraFirewall{storageDirectory:storageDirectory}, nil
+}
+
+
+func (firewall *FilesystemAcraFirewall) AddSpecificHandler(handler QueryHandler){
+
+	firewall.handlers = append(firewall.handlers, handler)
+}
+
+func (firewall *FilesystemAcraFirewall) HandleQuery(query string) error{
+
+	if len(firewall.handlers) < 1 {
```

Comment:
Firewall may contain 0 handlers if user will turn off all rules. As I remember we allow to configure it and leave the ability to run without any checks. If so we don't need this check and error.

Reply:
Agree. Removed